### PR TITLE
ar71xx : add support for D-Link DAP-1533

### PR DESCRIPTION
**Pull request by Tarask1n**
Access point D-Link DAP-1533:
- 4x 1Gbit port switch
- 1x 450Mbit 2.4/5GHz WiFi 
- 1x USB 2.0 port

It based on DB120 board:
- SoC Atheros AR7370 (AR9344 w/o WiFi) 
- 8 MiB flash
- 128 MiB ram 
- Ethernet switch AR8327N
- WiFi 2.4/5GHz AR9380

It could be easy converted to 1GBit router with LEDE.
More detailed info I described there https://wiki.openwrt.org/toh/d-link/d-link_dap-1533_a1.

All works fine LED, buttons, Ethernet, WiFi, USB. But I've faced an issue. When I flashed an image to 16MiB flash the available space for software still left 3.7 Mb as an 8MiB flash. May I've missed something in configuration.

Thanks in advance.

Signed-off-by: Taras Gavrylenko gavrylenko.taras@gmail.com

### CODE REVIEW

File: target/linux/ar71xx/base-files/etc/diag.sh [hunk 1]
```diff
@@ -142,7 +142,8 @@ get_status_led() {
 	dir-600-a1|\
 	dir-615-e1|\
 	dir-615-i1|\
-	dir-615-e4)
+	dir-615-e4|\
+	dap-1533-a1)
```
Comment (stintel) [inline target/linux/ar71xx/base-files/etc/diag.sh]: Please maintain alphabetical order.

File: target/linux/ar71xx/files/arch/mips/ath79/mach-dap-1533-a1.c [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+	},
+};
+
+static struct flash_platform_data dap1533a1_flash_data = {
+	.type = "mx25l6406e",
+};
+
+static void __init dap1533a1_setup(void)
+{
+//	u8 *mac = (u8 *) KSEG1ADDR(0x1ffe0000);
```
Comment (stintel) [inline target/linux/ar71xx/files/arch/mips/ath79/mach-dap-1533-a1.c]: Remove this if it isn't used.

**Comment by Tarask1n** [comment]
Requested chamges were fixed.

**Comment by pepe2k** [comment]
Hello @Tarask1n,

Please:
1. Squash all your commits into single one.
2. Move out image generation code from legacy.mk.
3. Drop executable file attribute on source files (don't change this, please).

Let me know when you fix all issues.

Cheers,
Piotr

**Comment by Tarask1n** [comment]
Hello Piotr,

Sorry for delayed reply.

1) Commits squashed
2) Legacy removed
3) Executable attribute switched off

BR
Taras.

**Comment by blogic** [comment]
With the remerge in progress, all PRs on the lede-project organisation will be closed. Please help getting this merged or rebase/post it on the openwrt project page (https://github.com/openwrt/openwrt/pulls). All remaining PRs will be closed in 30 days.

**Comment by pepe2k** [comment]
Superseded by https://github.com/openwrt/openwrt/pull/623.

**Pull request closed without merging**